### PR TITLE
Update branch cutting instructions.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Add instructions for creating dev tag.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -2,10 +2,50 @@
...
+   
+To create an empty commit and tag on master; run the following commands.  This will push directly to master,
+bypassing the normal PR process.  This is important to make sure that the tag is directly on the master branch.
+We create an empty commit because, when the release branch is created, it shares its commit history with master.
+So, if we tagged the tip of master, we'd also be tagging the tip of the release branch, which would give 
+incorrect results for `git describe --tags` on the release branch.
+
+   ```sh
+   git checkout <remote>/master
+   git commit --allow-empty -m "Start development on vX.Y"  # Where vX.Y is the next minor version
```
Comment (tmjd) [inline RELEASING.md:38]: Why do we need this commit? Why not just put a tag on the most recent commit?

Comment (fasaxc) [reply RELEASING.md:38]: That is explained in the paragraph immediately above ;-)

**Review by tmjd — APPROVED** [review]
LGTM, lets wait for Dan to review before merging.

**Review by danudey — APPROVED** [review]
lgtm

**Pull request merged**